libs/repositories/src/repos/TokenBalancesRepository/TokenBalancesRepositoryAlchemy.ts

```diff
@@ -0,0 +1,143 @@
+import { injectable } from 'inversify';
+import { TokenBalanceParams, TokenBalancesRepository, TokenBalancesResponse } from './TokenBalancesRepository';
+import { ALCHEMY_API_KEY, ALCHEMY_CLIENT_NETWORK_MAPPING, getAlchemyApiUrl } from '../../datasources/alchemy';
+import { NATIVE_CURRENCY_ADDRESS, ZERO_ADDRESS } from '@cowprotocol/cow-sdk';
+import { logger } from '@cowprotocol/shared';
+
+const JSON_RPC_VERSION = '2.0';
+const JSON_RPC_REQUEST_ID = 1;
+const REQUEST_TIMEOUT_MS = 10_000;
+const TOKEN_SPEC = ['erc20', 'NATIVE_TOKEN'] as const;
+
+type AlchemyTokenBalance = {
+  contractAddress: string | null;
+  tokenBalance: string;
+  error?: string;
+};
+
+type AlchemyGetTokenBalancesResponse = {
+  jsonrpc: string;
+  id: number;
+  result: {
+    address: string;
+    tokenBalances: AlchemyTokenBalance[];
+    pageKey?: string;
+  };
+};
+
+function isAlchemyGetTokenBalancesResponse(
+  data: unknown
+): data is AlchemyGetTokenBalancesResponse {
+  if (!data || typeof data !== 'object') {
+    return false;
+  }
+
+  const response = data as AlchemyGetTokenBalancesResponse;
+
+  if (
+    response.jsonrpc !== JSON_RPC_VERSION ||
+    typeof response.id !== 'number' ||
+    !response.result ||
+    typeof response.result !== 'object'
+  ) {
+    return false;
+  }
+
+  return Array.isArray(response.result.tokenBalances);
+}
+
+@injectable()
+export class TokenBalancesRepositoryAlchemy implements TokenBalancesRepository {
+  async getTokenBalances({
```

Comment:
Question: is there a HTTP cache for the request? I think it at least 5 seconds cache won't harm